test/e2e/preset_test.go

```diff
@@ -608,6 +617,122 @@ func validateWorkspaceReadiness(workspaceObj *kaitov1alpha1.Workspace) {
 	})
 }
 
+// Retry function with exponential backoff
+func retryRequest(req *http.Request, client *http.Client) (*http.Response, error) {
+	var resp *http.Response
+	var err error
+	backoff := initialBackoff
+
+	for i := 0; i < maxRetries; i++ {
+		resp, err = client.Do(req)
+		if err == nil && resp.StatusCode == http.StatusOK {
+			return resp, nil
+		}
+
+		if resp != nil {
+			resp.Body.Close()
+		}
+
+		fmt.Printf("Request failed (attempt %d/%d), retrying in %v...\n", i+1, maxRetries, backoff)
+		time.Sleep(backoff)
+		backoff *= 2
+		if backoff > maxBackoff {
+			backoff = maxBackoff
+		}
+	}
+	return nil, fmt.Errorf("request failed after %d retries: %v", maxRetries, err)
+}
+
+func validateModelsEndpoint(workspaceObj *kaitov1alpha1.Workspace) {
+	By("Validating the /v1/models endpoint", func() {
+		serviceURL := fmt.Sprintf("http://%s.%s.svc.cluster.local:80/v1/models",
+			workspaceObj.Name, namespaceName)
+
+		// Get the model name directly from workspaceObj
+		if workspaceObj.Inference == nil || workspaceObj.Inference.Preset == nil {
+			Fail(fmt.Sprintf("No preset inference model found in workspace %s", workspaceObj.Name))
+		}
+		modelName := workspaceObj.Inference.Preset.Name // Dynamically retrieved
+
+		req, err := http.NewRequest("GET", serviceURL, nil)
```

Comment:
can we abstract this http operation to a method to dedup code? 
doHTTPGet(url string) ([]byte, error)